src/Symfony/Component/Form/Extension/PasswordHasher/EventListener/PasswordHasherListener.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\PasswordHasher\EventListener;
+
+use Symfony\Component\Form\Exception\InvalidConfigurationException;
+use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
+
+/**
+ * @author Sébastien Alfaiate <s.alfaiate@webarea.fr>
+ */
+class PasswordHasherListener
+{
+    private UserPasswordHasherInterface $passwordHasher;
+    private PropertyAccessorInterface $propertyAccessor;
+
+    private array $passwords = [];
+
+    public function __construct(UserPasswordHasherInterface $passwordHasher, PropertyAccessorInterface $propertyAccessor = null)
+    {
+        $this->passwordHasher = $passwordHasher;
+        $this->propertyAccessor = $propertyAccessor ?? PropertyAccess::createPropertyAccessor();
+    }
+
+    public function registerPassword(FormEvent $event)
+    {
+        $form = $event->getForm();
+        $parentForm = $form->getParent();
+        $mapped = $form->getConfig()->getMapped();
+
+        if ($parentForm && $parentForm->getConfig()->getType()->getInnerType() instanceof RepeatedType) {
+            $mapped = $parentForm->getConfig()->getMapped();
+            $parentForm = $parentForm->getParent();
+        }
+
+        if ($mapped) {
+            throw new InvalidConfigurationException('The "hash_property_path" option cannot be used on mapped field.');
+        }
+
+        if (!$parentForm || !($user = $parentForm->getData()) || !($user instanceof PasswordAuthenticatedUserInterface)) {
+            throw new InvalidConfigurationException(sprintf('The "hash_property_path" option only supports "%s" objects.', PasswordAuthenticatedUserInterface::class));
+        }
+
+        $this->passwords[] = [
+            'user' => $user,
+            'property_path' => $form->getConfig()->getOption('hash_property_path'),
+            'hashed_password' => $this->passwordHasher->hashPassword($user, $event->getData()),
```

Comment:
I would move the hashing process after we know the form is valid, that would save some ticks and improve the performance a bit when the form is not valid.

Reply:
Done: https://github.com/symfony/symfony/pull/46224/commits/5afae70045fc3829a9bf7162dfb60295381913ac